src/lang_items.rs

```diff
@@ -19,36 +19,27 @@
 //! crate.
 
 use crate::led;
-use crate::result::TockError;
 use crate::timer;
 use crate::timer::Duration;
 use core::alloc::Layout;
 use core::executor;
-use core::future::Future;
 use core::panic::PanicInfo;
 
 #[lang = "start"]
 extern "C" fn start<T>(main: fn() -> T, _argc: isize, _argv: *const *const u8) -> i32
```

Comment:
I learned from @jrvanwhy that you can even use the unit type here.

Reply:
When I change this I get:
```
error: internal compiler error: src/librustc_metadata/decoder.rs:931: get_optimized_mir: missing MIR for `DefId(11:681 ~ libtock[c3e0]::lang_items[0]::start[0])`
```